pilot/pkg/networking/core/v1alpha3/gateway.go

```diff
@@ -437,6 +454,54 @@ func buildGatewayListenerTLSContext(server *networking.Server, enableSds bool) *
 				},
 			}
 		}
+	} else if server.Tls.Mode == networking.Server_TLSOptions_ISTIO_MUTUAL {
+		// global SDS must be enabled if the sds path is non-default, configure TLS with SDS
+		// TODO would prefer something less implicit here
+		if sdsConfig.path != "" {
+			// configure egress with SDS
+			tls.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_CombinedValidationContext{
+				CombinedValidationContext: &auth.CommonTlsContext_CombinedCertificateValidationContext{
+					DefaultValidationContext: &auth.CertificateValidationContext{VerifySubjectAltName: server.Tls.SubjectAltNames},
+					ValidationContextSdsSecretConfig: authn_model.ConstructSdsSecretConfig(
+						authn_model.SDSRootResourceName, sdsConfig.path, sdsConfig.enableSdsTokenMount, sdsConfig.sdsUseK8sJwt, sdsConfig.metadata),
+				},
+			}
+			tls.CommonTlsContext.TlsCertificateSdsSecretConfigs = []*auth.SdsSecretConfig{
+				authn_model.ConstructSdsSecretConfig(
+					authn_model.SDSDefaultResourceName, sdsConfig.path, sdsConfig.enableSdsTokenMount, sdsConfig.sdsUseK8sJwt, sdsConfig.metadata)}
+		} else {
+			// global SDS disabled, fall back on using mounted certificates
+			caCertificates := model.GetOrDefaultFromMap(sdsConfig.metadata, model.NodeMetadataTLSClientRootCert, "/etc/certs/root-cert.pem")
+			trustedCa := &core.DataSource{
+				Specifier: &core.DataSource_Filename{
+					Filename: caCertificates,
+				},
+			}
+			tls.CommonTlsContext.ValidationContextType = &auth.CommonTlsContext_ValidationContext{
+				ValidationContext: &auth.CertificateValidationContext{
+					TrustedCa:            trustedCa,
+					VerifySubjectAltName: server.Tls.SubjectAltNames,
+				},
+			}
+
+			// each of the default paths should be in a constants section somewhere
```

Comment:
yes :)

grep the code base or define it somewhere if not exists.

Reply:
Just looked around, seems like the value is hardcoded into flag defaults for the most part, I'll add a common place for these as part of this PR

[https://github.com/istio/istio/blob/93df5e4b38400fdd9d2fd0fe38abf8400532b8d9/galley/cmd/galley/cmd/server.go#L103](https://github.com/istio/istio/blob/93df5e4b38400fdd9d2fd0fe38abf8400532b8d9/galley/cmd/galley/cmd/server.go#L103)
[https://github.com/istio/istio/blob/93df5e4b38400fdd9d2fd0fe38abf8400532b8d9/security/cmd/node_agent/main.go#L66](https://github.com/istio/istio/blob/93df5e4b38400fdd9d2fd0fe38abf8400532b8d9/security/cmd/node_agent/main.go#L66)
[https://github.com/istio/istio/blob/93df5e4b38400fdd9d2fd0fe38abf8400532b8d9/galley/pkg/crd/validation/webhook.go#L160](https://github.com/istio/istio/blob/93df5e4b38400fdd9d2fd0fe38abf8400532b8d9/galley/pkg/crd/validation/webhook.go#L160)
[

Reply:
How about `pkg/config/constants.go`?

Reply:
https://github.com/istio/istio/blob/master/pkg/config/constants/constants.go#L45

Reply:
Ah, I was grepping for `/etc/certs/cert-chain` and missed it. Will use those constants :+1: